src/main/java/org/example/springbootintro/repository/book/BookSpecificationBuilder.java

```diff
@@ -0,0 +1,33 @@
+package org.example.springbootintro.repository.book;
+
+import lombok.AllArgsConstructor;
+import org.example.springbootintro.dto.book.BookSearchParametersDto;
+import org.example.springbootintro.model.Book;
+import org.example.springbootintro.repository.SpecificationBuilder;
+import org.example.springbootintro.repository.SpecificationProviderManager;
+import org.example.springbootintro.repository.book.spec.AuthorSpecificationProvider;
+import org.example.springbootintro.repository.book.spec.TitleSpecificationProvider;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@AllArgsConstructor
```

Comment:
```suggestion
@RequiredArgsConstructor
```
as you wanna have only required properties injected

Reply:
https://github.com/EduarfF/book-store/pull/6/files#r1585980458
not fixed